forge/src/main/java/org/openjdk/skara/forge/github/GitHubRepository.java

```diff
@@ -263,18 +260,49 @@ public VCS repositoryType() {
 
     @Override
     public Optional<String> fileContents(String filename, String ref) {
-        var content = request.get("contents/" + filename)
-                .param("ref", ref)
-                .onError(r -> r.statusCode() == 404 && JSON.parse(r.body()).get("message").asString().equals("Not Found")
-                        ? Optional.of(JSON.object().put("NOT_FOUND", true)) : Optional.empty())
+        // Get file contents using raw format. This allows us to get files of
+        // size up to 10MB (up from 1MB if getting in object from).
+        try {
+            var content = request.get("contents/" + filename)
+                    .param("ref", ref)
+                    .header("Accept", "application/vnd.github.raw+json")
+                    .executeUnparsed();
+            return Optional.of(content);
+        } catch (UncheckedRestException e) {
+            // The onError handler is not used with executeParsed, so have to
+            // resort to catching exception for 404 handling.
+            if (e.getStatusCode() == 404) {
+                return Optional.empty();
+            } else {
+                throw e;
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void writeFileContents(String content, String filename, Branch branch, String message, String authorName, String authorEmail) {
+        var body = JSON.object()
+                .put("message", message)
+                .put("branch", branch.name())
+                .put("committer", JSON.object()
+                        .put("name", authorName)
+                        .put("email", authorEmail))
+                .put("content", new String(Base64.getEncoder().encode(content.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8));
```

Comment:
Are we talking about 10's of MB here as well? We don't risk running into limits?

Reply:
Hm, I just realized the comment on fileContents is wrong, with raw we get up to 100MB, so I will fix that. I can't find any documentation regarding a limit when writing file contents. The manual test I have tests up to 10MB, which was fine. The biggest file in our current archive is just over 1MB. The current archive is however hosted on GitLab and not GitHub, so the GitHub limits aren't really relevant at this time.